CONTRIBUTING.md

```diff
@@ -15,39 +15,141 @@ Otherwise, go read Docker's
 
 ### Development Environment Setup
 
-Swarm is written in [the Go programming language](http://golang.org) and manages its dependencies using [Godep](http://github.com/tools/godep). Go and Godep also use `git` to fetch dependencies.
-So, you are required to install the Go compiler, and the `git` client (e.g. `apt-get install golang git` on Ubuntu).
+Swarm is written in [the Go programming language](http://golang.org) and manages its dependencies using [Godep](http://github.com/tools/godep).  This guide will walk you through installing Go, forking the Swarm repo, building Swarm from source and contributing pull requests (PRs) back to the Swarm project.
 
-You may need to setup your path and set the `$GOPATH` and `$GOBIN` variables, e.g:
+#### Install git, Go and Godep
+If you don't already have `git` installed, you should install it.  For example, on Ubuntu:
 ```sh
-mkdir -p ~/projects/swarm
-cd ~/projects/swarm
-export GOPATH=$PWD
-export GOBIN=$GOPATH/bin
+sudo apt-get install git
 ```
 
-As previously mentioned, you need `godep` to manage dependencies for Swarm. The `godep` command can be installed using the following command:
+You also need Go 1.4+.  Download Go from [https://golang.org/dl/](https://golang.org/dl/).  To install on Linux:
```

Comment:
Nevermind, @MHBauer , I see now you are correct that we require Go 1.5(.2) or higher.

I only just now discovered your PR #1190.  Sorry for not noticing that sooner.  My proposal is that we not have build instructions in the README.md, but instead use README.md as a kind of homepage for the repo (since it's the thing that gets displayed when a person browses to github.com/docker/swarm).

Maybe it is a mistake to conflate build instructions with a how-to-contribute guide in CONTRIBUTING.md.  I am not sure.  What do you think?